core/src/main/java/org/apache/iceberg/GenericBlobMetadata.java

```diff
@@ -26,7 +27,17 @@
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
 import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
 
-public class GenericBlobMetadata implements BlobMetadata {
+public class GenericBlobMetadata implements BlobMetadata, Serializable {
+
+  public static BlobMetadata from(org.apache.iceberg.puffin.BlobMetadata puffinMetadata) {
```

Comment:
Why don't we make it so that the Puffin `BlobMetadata` implements `BlobMetadata`? Then this would be unnecessary.

Reply:
Puffin BlobMetadata contains all the blob metadata information from the Puffin footer.
In particular, it contains more fields, like offset and length, which are not carried over to the table metadata.

We can make Puffin `BlobMetadata` implement the `BlobMetadata`, but
- we don't want to serialize (as in `Serializable`) the Puffin BlobMetadata, as we're not interested in sending over those additional fields (I know these are small today, and _maybe_ are never big in the future)
- there is a challenge with equality semantics. `GenericBlobMetadata` implements equals, but with more complicated class hierarchy it wouldn't be useful anymore.

Let me know what you think

Reply:
> we don't want to serialize (as in Serializable) the Puffin BlobMetadata

i see https://github.com/apache/iceberg/pull/4741#discussion_r966204993 now